core/src/main/scala/org/apache/spark/util/ListenerBus.scala

```diff
@@ -111,7 +118,10 @@ private[spark] trait ListenerBus[L <: AnyRef, E] extends Logging {
           logError(s"Listener ${Utils.getFormattedClassName(listener)} threw an exception", e)
       } finally {
         if (maybeTimerContext != null) {
-          maybeTimerContext.stop()
+          val elapsed = maybeTimerContext.stop()
+          if (logSlowEventEnabled && elapsed > logSlowEventThreshold) {
+            logError(s"Process of event ${event} took ${elapsed / 1000000000d}s.")
```

Comment:
My 2 cents, given it doesn't make things crash, warning might be better.
And logging which listener is important as the problematic one might not be specific event but might be specific listener. It could be a combination of (specific listener, specific event).

Reply:
updated